src/main/flight/failsafe.c

```diff
@@ -377,7 +379,16 @@ void failsafeUpdateState(void)
                 if (receivingRxData) {
                     failsafeState.phase = FAILSAFE_RX_LOSS_RECOVERED;
                 } else {
-                    switch (failsafeConfig()->failsafe_procedure) {
+                    uint8_t failsafe_procedure_to_use = failsafeConfig()->failsafe_procedure;
+
+                    // Craft is closer than minimum failsafe procedure distance (if set to non-zero)
+                    if ((failsafeConfig()->failsafe_min_distance > 0) && 
+                        (GPS_distanceToHome < failsafeConfig()->failsafe_min_distance)) {
```

Comment:
This logic needs to check if home position is set at all. We also need to decide what procedure to execute if GPS is not available, or we took off earlier than GPS was fixed and home position recorded.

Reply:
That's a very good point. I've made this change, thank you.